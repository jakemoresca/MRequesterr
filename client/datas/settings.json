{
    "integrationSettings": {
        "movies" : {
            "baseUrl": "",
            "apiKey": "",
            "host": "",
            "port": 80,
            "useSsl": false
        },
        "series" : {
            "baseUrl": "",
            "apiKey": "",
            "host": "",
            "port": 80,
            "useSsl": false
        },
        "tmdbApiKey": ""
    }
}