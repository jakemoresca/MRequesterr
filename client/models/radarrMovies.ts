export interface IRadarrQueue {
    records: RadarrQueueRecord[];
}

export interface RadarrQueueRecord {
    movieId:                 number;
    languages:               Language[];
    quality:                 RecordQuality;
    customFormats:           any[];
    size:                    number;
    title:                   string;
    sizeleft:                number;
    timeleft:                string;
    estimatedCompletionTime: Date;
    status:                  string;
    trackedDownloadStatus:   string;
    trackedDownloadState:    string;
    statusMessages:          any[];
    downloadId:              string;
    protocol:                string;
    downloadClient:          string;
    indexer:                 string;
    outputPath:              string;
    id:                      number;
}

export interface Language {
    id:   number;
    name: string;
}

export interface RecordQuality {
    quality:  QualityQuality;
    revision: Revision;
}

export interface QualityQuality {
    id:         number;
    name:       string;
    source:     string;
    resolution: number;
    modifier:   string;
}

export interface Revision {
    version:  number;
    real:     number;
    isRepack: boolean;
}
