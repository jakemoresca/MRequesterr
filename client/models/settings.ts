export interface ISettings
{
    integrationSettings: IIntegrationSettings;
}

export interface IIntegrationSettings
{
    movies: IMovieSettings;
    series: ISeriesSettings;
}

export interface IMovieSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface ISeriesSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}