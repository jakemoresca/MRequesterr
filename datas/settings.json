{
    "integrationSettings": {
        "movies": {
            "baseUrl": "/radarr",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "series": {
            "baseUrl": "/sonarr",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "auth": {
            "baseUrl": "/jellyfin",
            "apiKey": "",
            "host": "rydersir.polaris.usbx.me",
            "port": 443,
            "useSsl": true
        },
        "tmdbApiKey": ""
    },
    "plexLogin": true
}