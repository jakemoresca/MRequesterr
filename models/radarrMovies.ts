export interface IRadarrQueue {
    records: RadarrQueueRecord[];
}

export interface RadarrQueueRecord {
    movieId: number;
    languages: Language[];
    quality: RecordQuality;
    customFormats: any[];
    size: number;
    title: string;
    sizeleft: number;
    timeleft: string;
    estimatedCompletionTime: Date;
    status: string;
    trackedDownloadStatus: string;
    trackedDownloadState: string;
    statusMessages: any[];
    downloadId: string;
    protocol: string;
    downloadClient: string;
    indexer: string;
    outputPath: string;
    id: number;
}

export interface Language {
    id: number;
    name: string;
}

export interface RecordQuality {
    quality: QualityQuality;
    revision: Revision;
}

export interface QualityQuality {
    id: number;
    name: string;
    source: string;
    resolution: number;
    modifier: string;
}

export interface Revision {
    version: number;
    real: number;
    isRepack: boolean;
}

export interface IRadarrMovie {
    title: string;
    originalTitle: string;
    originalLanguage: Language;
    alternateTitles: AlternateTitle[];
    secondaryYearSourceId: number;
    sortTitle: string;
    sizeOnDisk: number;
    status: string;
    overview: string;
    inCinemas: Date;
    physicalRelease: Date;
    digitalRelease: Date;
    images: Image[];
    website: string;
    remotePoster: string;
    year: number;
    hasFile: boolean;
    youTubeTrailerId: string;
    studio: string;
    qualityProfileId: number;
    monitored: boolean;
    minimumAvailability: string;
    isAvailable: boolean;
    folderName: string;
    runtime: number;
    cleanTitle: string;
    imdbId: string;
    tmdbId: number;
    titleSlug: string;
    folder: string;
    certification: string;
    genres: string[];
    tags: any[];
    added: Date;
    ratings: Ratings;
    id: number;
    addOptions: AddOptions;
    rootFolderPath: string;
    apikey: string;
}

export interface AddOptions {
    searchForMovie: boolean;
}

export interface AlternateTitle {
    sourceType: SourceType;
    movieId: number;
    title: string;
    sourceId: number;
    votes: number;
    voteCount: number;
    language: Language;
}

export interface Language {
    id: number;
    name: string;
}

export enum SourceType {
    Tmdb = "tmdb",
}

export interface Image {
    coverType: string;
    url: string;
    remoteUrl: string;
}

export interface Ratings {
    imdb: Imdb;
    tmdb: Imdb;
    metacritic: Imdb;
    rottenTomatoes: Imdb;
}

export interface Imdb {
    votes: number;
    value: number;
    type: string;
}

