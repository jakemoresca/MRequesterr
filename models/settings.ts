export interface ISettings
{
    integrationSettings: IIntegrationSettings;
    plexLogin: boolean;
}

export interface IIntegrationSettings
{
    movies: IMovieSettings;
    series: ISeriesSettings;
    auth: IAuthSettings;
}

export interface IMovieSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface ISeriesSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}

export interface IAuthSettings
{
    baseUrl: string;
    apiKey: string;
    host: string;
    port?: number;
    useSsl: boolean;
}