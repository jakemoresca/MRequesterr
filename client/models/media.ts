import { IRadarrMovie, RadarrQueueRecord } from "./radarrMovies";
import { ISonarrSeries } from "./sonarrSeries";

export interface IMedia 
{
    id: string;
    tmdbId?: string;
    imdbId: string;
    tvdbId?: string;
    title: string;
    cleanTitle: string;
    sortTitle: string;
    status: string;
    overview: string;
    monitored: boolean;
    minimumAvailability: string;
    //profileId: number;
    runtime: number;
    images: IMediaImage[];
    titleSlug: string;
    path: string;
    year: number;
    genres: string[];
    isAvailable: boolean;
    hasFile: boolean;
    statistics: IStatistics;
    additionalInfo?: AdditionalMediaInfo;
    movieFile?: MovieFile;
    firstAired?: string;
}

export type AdditionalMediaInfo = ISonarrSeries | RadarrQueueRecord | IRadarrMovie | undefined;

export interface IMediaImage 
{
    coverType: string;
    url: string;
}

export interface IStatistics 
{
    percentOfEpisodes: number;
}

export interface MovieFile {
    movieId:             number;
    relativePath:        string;
    path:                string;
    size:                number;
    dateAdded:           Date;
    indexerFlags:        number;
    quality:             Welcome4Quality;
    mediaInfo:           MediaInfo;
    originalFilePath:    string;
    qualityCutoffNotMet: boolean;
    languages:           Language[];
    releaseGroup:        string;
    edition:             string;
    id:                  number;
}

export interface Language {
    id:   number;
    name: string;
}

export interface MediaInfo {
    audioBitrate:          number;
    audioChannels:         number;
    audioCodec:            string;
    audioLanguages:        string;
    audioStreamCount:      number;
    videoBitDepth:         number;
    videoBitrate:          number;
    videoCodec:            string;
    videoDynamicRangeType: string;
    videoFps:              number;
    resolution:            string;
    runTime:               string;
    scanType:              string;
    subtitles:             string;
}

export interface Welcome4Quality {
    quality:  QualityQuality;
    revision: Revision;
}

export interface QualityQuality {
    id:         number;
    name:       string;
    source:     string;
    resolution: number;
    modifier:   string;
}

export interface Revision {
    version:  number;
    real:     number;
    isRepack: boolean;
}
