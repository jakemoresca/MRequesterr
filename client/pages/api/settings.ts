// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
import type { NextApiRequest, NextApiResponse } from 'next'
import { ISettings } from '../../models/settings'

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ISettings>
) {
    const settings = await getSettings();
    res.status(200).json(settings)
}

export async function getSettings(): Promise<ISettings> {
    const settings = await import("../../datas/settings.json");

    settings.integrationSettings.movies.apiKey = process.env.RADARR_API_KEY ?? settings.integrationSettings.movies.apiKey;
    settings.integrationSettings.series.apiKey = process.env.SONARR_API_KEY ?? settings.integrationSettings.series.apiKey;
    settings.integrationSettings.auth.apiKey = process.env.JELLYFIN_API_KEY ?? settings.integrationSettings.auth.apiKey;

    return settings.default;
}
