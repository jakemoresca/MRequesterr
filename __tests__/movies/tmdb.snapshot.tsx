import { render } from '@testing-library/react'
import React from 'react';
import { RecoilRoot } from 'recoil';
import { IRadarrMovie, SourceType } from '../../models/radarrMovies';
import Movie from '../../pages/movies/[tmdbId]';
import { MediaStateType } from '../../states/media';
import { ISettings } from '../models/settings'

jest.mock("../../components/authenticate", () => (props) => {
  return (<mock-authenticate data-testid="authenticate">{props.children}</mock-authenticate>);
});

const media: MediaStateType = {
  id: "",
  tmdbId: "",
  overview: "",
  imdbId: "",
  cleanTitle: "",
  sortTitle: "",
  title: "",
  titleSlug: "",
  status: "",
  monitored: false,
  minimumAvailability: "",
  runtime: 0,
  images: [{ coverType: "poster", url: "" }],
  path: "",
  year: 1,
  genres: [],
  isAvailable: false,
  hasFile: false,
  statistics: {
    percentOfEpisodes: 0
  }
}

const radarrLookup: IRadarrMovie = {
  ...media,
  originalTitle: "",
  originalLanguage: { id: 0, name: "" },
  alternateTitles: [{ sourceType: SourceType.Tmdb, movieId: 0, title: "", sourceId: 0, votes: 0, voteCount: 0, language: { id: 0, name: "" } }],
  secondaryYearSourceId: 0,
  sizeOnDisk: 0,
  inCinemas: new Date,
  physicalRelease: new Date,
  digitalRelease: new Date,
  website: "",
  remotePoster: "",
  youTubeTrailerId: "",
  studio: "",
  qualityProfileId: 0,
  folderName: "",
  folder: "",
  certification: "",
  tags: [],
  added: new Date,
  addOptions: { searchForMovie: false },
  rootFolderPath: "",
  apikey: "",
  tmdbId: 0,
  id: 0
}

jest.mock('../../pages/api/movies', () => {
  return {
    getMovies: () => [],
    getQueue: () => [],
    getPopularSeries: () => ({ results: [] }),
    requestMovie: () => { },
    getMovieLookup: () => radarrLookup
  }
})

jest.mock('../../pages/api/tmdb', () => {
  return {
    convertToMedia: () => media,
    getMovie: () => { }
  }
})

jest.mock('next/router', () => ({
  useRouter() {
    return ({
      query: {
        tmdbId: ""
      },
    });
  },
}));

it('renders movie page unchanged', () => {
  const settings: ISettings = {
    "integrationSettings": {
      "movies": {
        "baseUrl": "/radarr",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      },
      "series": {
        "baseUrl": "/sonarr",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      },
      "auth": {
        "baseUrl": "/jellyfin",
        "apiKey": "",
        "host": "test.web.com",
        "port": 443,
        "useSsl": true
      }
    }
  }

  const { container } = render(<RecoilRoot override={true}><Movie settings={settings} /></RecoilRoot>)
  expect(container).toMatchSnapshot()
});