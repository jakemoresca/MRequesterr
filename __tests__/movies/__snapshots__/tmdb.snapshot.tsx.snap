// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie page unchanged 1`] = `
<div>
  <div
    class="py-3 container-fluid"
  >
    <mock-authenticate
      data-testid="authenticate"
    >
      <div
        class="bg-dark col-12 card text-white border-dark"
      >
        <div
          class="row g-0"
        >
          <div
            class="col-md-2"
          >
            <img
              class="rounded-start w-auto card-img-top"
              height="368"
              width="250"
            />
          </div>
          <div
            class="col-md-10"
          >
            <div
              class="card-body"
            >
              <h2
                class="card-title"
              />
              <h6
                class="mb-2 text-muted card-subtitle"
              />
              <h6
                class="mb-2 text-muted card-subtitle"
              />
              <button
                class="btn btn-primary"
                type="button"
              >
                Request
              </button>
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        class="d-flex flex-row container-fluid"
      />
    </mock-authenticate>
  </div>
</div>
`;
